src/libraries/System.Reflection.Metadata/tests/Metadata/TypeSystem/TypeDefinitionTests.cs

```diff
@@ -32,5 +33,45 @@ public void ValidateTypeDefinitionIsNestedWindowsProjection()
                 Assert.Equal(typeDef.Attributes.IsNested(), typeDef.IsNested);
             }
         }
+
+        [Fact]
+        public void ValidateDeclaringType()
+        {
+            var reader = MetadataReaderTests.GetMetadataReader(Misc.Members);
+
+            Assert.NotEmpty(reader.GetTypesWithEvents());
+            Assert.NotEmpty(reader.GetTypesWithProperties());
+
+            foreach (var typeDefHandle in reader.TypeDefinitions)
+            {
+                var typeDef = reader.GetTypeDefinition(typeDefHandle);
+                // The assembly does not declare any nested types yet.
```

Comment:
Would it be better to add a nested type to this assembly instead?

Reply:
Added a test.